cmd/clusterctl/pkg/log/logger_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package log
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+func TestFlatten(t *testing.T) {
+	type args struct {
+		prefix string
+		kvList []interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "message without values",
+			args: args{
+				prefix: "",
+				kvList: []interface{}{
+					"msg", "this is a message",
+				},
+			},
+			want: "this is a message",
+		},
+		{
+			name: "message with values",
+			args: args{
+				prefix: "",
+				kvList: []interface{}{
+					"msg", "this is a message",
+					"val1", 123,
+				},
+			},
+			want: "this is a message val1=123",
```

Comment:
if val1's value is has whitespace in it, e.g. `foo bar`, and let's say you have another kv pair, do you get `this is a message val1=foo bar val2=hello world`? If so, let's bring back quotes. Or I guess the `pretty` call to json marshal will put the quotes in?

Reply:
If the value is a string it gets surrounded by quotes val1="foo bar".
the quotes are dropped only from the key

Reply:
Could you add a test case that has a string and/or multi-word value?

Reply:
done!